wallet/src/main/java/com/android/identity_credential/wallet/SelfSignedMdocIssuingAuthority.kt

```diff
@@ -0,0 +1,235 @@
+import android.graphics.Shader
+import androidx.annotation.RawRes
+import com.android.identity.cbor.Bstr
+import com.android.identity.cbor.Cbor
+import com.android.identity.cbor.DataItem
+import com.android.identity.cbor.Tagged
+import com.android.identity.cbor.toDataItem
+import com.android.identity.cose.Cose
+import com.android.identity.cose.CoseLabel
+import com.android.identity.cose.CoseNumberLabel
+import com.android.identity.crypto.Algorithm
+import com.android.identity.crypto.Certificate
+import com.android.identity.crypto.CertificateChain
+import com.android.identity.crypto.EcPrivateKey
+import com.android.identity.crypto.EcPublicKey
+import com.android.identity.crypto.javaX509Certificate
+import com.android.identity.issuance.CredentialConfiguration
+import com.android.identity.issuance.CredentialPresentationFormat
+import com.android.identity.issuance.simple.SimpleIssuingAuthority
+import com.android.identity.mdoc.mso.MobileSecurityObjectGenerator
+import com.android.identity.mdoc.mso.StaticAuthDataGenerator
+import com.android.identity.mdoc.util.MdocUtil
+import com.android.identity.storage.StorageEngine
+import com.android.identity.util.Logger
+import com.android.identity.util.Timestamp
+import java.io.ByteArrayOutputStream
+import java.nio.charset.StandardCharsets
+import kotlin.math.ceil
+import kotlin.random.Random
+
+
+abstract class SelfSignedMdocIssuingAuthority(
+    val application: WalletApplication,
+    storageEngine: StorageEngine
+): SimpleIssuingAuthority(
+    storageEngine
+) {
+    companion object {
+        private const val TAG = "SelfSignedMdlIssuingAuthority"
+
+        val MDL_DOCTYPE = "org.iso.18013.5.1.mDL"
+        val MDL_NAMESPACE = "org.iso.18013.5.1"
+        val AAMVA_NAMESPACE = "org.iso.18013.5.1.aamva"
+    }
+
+    override fun createPresentationData(presentationFormat: CredentialPresentationFormat,
+                                        credentialConfiguration: CredentialConfiguration,
+                                        authenticationKey: EcPublicKey
+    ): ByteArray {
+        // Right now we only support mdoc
+        check(presentationFormat == CredentialPresentationFormat.MDOC_MSO)
+
+        val now = Timestamp.now()
+
+        // Create AuthKeys and MSOs, make sure they're valid for a long time
+        val timeSigned = now
+        val validFrom = now
+        val validUntil = Timestamp.ofEpochMilli(validFrom.toEpochMilli() + 365*24*3600*1000L)
+
+        // Generate an MSO and issuer-signed data for this authentication key.
+        val msoGenerator = MobileSecurityObjectGenerator(
+            "SHA-256",
+            MDL_DOCTYPE,
+            authenticationKey
+        )
+        msoGenerator.setValidityInfo(timeSigned, validFrom, validUntil, null)
+        val randomProvider = Random.Default
+        val issuerNameSpaces = MdocUtil.generateIssuerNameSpaces(
+            credentialConfiguration.staticData,
+            randomProvider,
+            16,
```

Comment:
Can we please have random numbers like this be either
- added as the default value for `generateIssuerNameSpaces()` function signature (so we don't have to specify this number), or
- explicitly label it `dataElementRandomSize = 16`, or
- store it in some const and reference it here `RANDOM_SEED_SIZE`